### PR TITLE
[DinoMod]Tweaked Spinosaurus and Ceratosaurus spawns/stats. Added NPC Hint.

### PR DESCRIPTION
**Pull request by jackledead**
#### Summary
SUMMARY: Mods "Spinosaurus lake spawns. Changed Spino stats and Spino/Ceratosaurus spawn weights. Added NPC Hint."
#### Purpose of change
New fossils have come out of Spinosaurus that give us details on how it moved in land as well as water and we now know Spinosaurus was an amphibious fish eater! So when I started playing this mod recently and saw Spinosaurus nowhere near lakes, you bet I was mad and hopped on it. In the process i found Ceratosauruses misbehaving by spawning in lakes more than mosasaurs were so I fixed that too. Added NPC hint because players need to know that the waters spawn scary dinos. While looking at the jsons i found that some of the dinos i were working on had ridiculously low spawn frequency (1) and i had never seen them in game, so i raised them a point or two here and there to make the mod more FUN and make the world feel more alive.

#### Describe the solution
Spinosaurus spawns tweaked to permit spawning in lakes, lightly tweaked Spinosaurus stats to more closely represent what we know about their fossils now, more hp and armor, slower speed, different calculation of melee dice, bashing damage (bite force and mouth size), lowered morale and aggression.
Tweaked Ceratosauruses to hopefully not spawn in the middle of large lakes anymore.
Increased spino frequency in some situations and lowered cost slightly (justification is their speed. they are  pretty easily avoidable in my testing on ground, but tougher in lakes, and easily killable with a gun of almost any caliber, even with their large size and hp).
Tweaked Zpinosaurus to spawn slightly more often later game.
Changed spinosaurus description.
Added NPC hint about mosasaurus/spinosaurus in water.

#### Describe alternatives you've considered
I tried to get spinosaurus to be able to swim faster in water but i couldnt find any sort of way to increase the swimming capabilities of monsters. I considered having two species of spino, one water and one near banks and beaches, but that would be a bit much i think.

#### Testing
Spawned as a class with a gun and basic armor (military recruit, bio weapon soldier) and debugged my athletics high enough to swim indefinitely and just kept swimming around the lake from the desert island wilderness spawn scenario. I got a gist of the spawn rates and how they display in game, was able to find a spino once in a blue moon in the lake (about 0.5 Spinos per sq km id say). I tweaked ceratosaurus until I stopped finding them in lakes with this method for the most part (id say one once in a while is okay, but not 4 in 5 minutes like it was before.)

I had combat with the spinos i found and spawned from debug and tweaked their damage speed and hp until it matched what i was able to research on Spinosaurus stats (remember, Spinosaurus is a genus not a species so stats vary).

Also wandered through various biomes to see spawn rates and was msotly pleased regarding ceratosaurus and spinosaurus by the end. 

#### Additional context
Fossils/videos/research
https://www.nationalgeographic.com/science/2020/04/first-spinosaurus-tail-found-confirms-dinosaur-was-swimming/
https://www.nationalgeographic.com/science/2020/04/spinosaurus-graphic-reconstructing-gigantic-aquatic-predator/
https://www.youtube.com/watch?v=De1VJ1UpjhM Spinosaurus 2020 overview
https://www.youtube.com/watch?v=6oCqSqY7FiI 2016 info on general skeleton as well as arm positioning and how it would have walked
https://youtu.be/jj8ZZNGWOhs spinosaurus general rundown in 2020, best video on subject

I plan to add more updates to every dino and maybe rebalance as a whole. I dont believe they currently represent their true force in life, and as it stands it's pretty easy to take out dinos with guns with how bleeding works, and i feel like dino bite forces should be similar to zombie brute/hulk/mi-go damage for the stronger dinos. Will investigate.

### CODE REVIEW

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 1]
```diff
@@ -131,24 +131,25 @@
     "weight": "1000 kg",
     "bodytype": "bear",
     "material": [ "flesh" ],
-    "aggression": 100,
-    "morale": 90,
-    "speed": 100,
+    "//": "Altering Spinosaurus to reflect modern fossils and research on speed, bite force, teeth shape, tail shape and size as well changing its aggression and morale to reflect our understanding of eating and hunting behaviours.",
```
Comment (LaVeyanFiend) [inline data/mods/DinoMod/monsters/dinosaur.json:134]: You probably want to add those fossils/research to the PR description.

Comment (jackledead) [reply data/mods/DinoMod/monsters/dinosaur.json:134]: done

File: data/mods/DinoMod/hints.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
       "I hear the zombies have been in the swamps.  Bad news if they bite a dinosaur before it bites them.",
       "I know there aren't alligators in the sewer, but I heard there was some kind of big lizard down there.  Probably not a good idea to check.",
       "Some of those big dinosaurs seem halfway all right.  I bet if you fed them something nice and gave them a pet you could ride them like a pony.  Or maybe they'd eat you instead.",
+      "I saw something with a large fin in the water. It was the size of a whale and gave me a menacing feeling. I'm gonna think twice before going back there.",
```
Comment (Qrox) [inline data/mods/DinoMod/hints.json:13]:
```suggestion
      "I saw something with a large fin in the water.  It was the size of a whale and gave me a menacing feeling.  I'm gonna think twice before going back there.",
```
Fixes the error in travis.

Comment (jackledead) [reply data/mods/DinoMod/hints.json]: Travis still rejecting after commit

File: data/mods/DinoMod/monsters/dinosaur.json [hunk 3]
```diff
@@ -131,24 +131,25 @@
...
     "armor_bash": 4,
-    "armor_cut": 2,
+    "armor_cut": 3,
     "armor_bullet": 2,
-    "hp": 400,
+    "hp": 425,
     "death_function": [ "NORMAL" ],
     "zombify_into": "mon_zpinosaurus",
-    "description": "A huge dinosaur about the size of a small house, with a ferocious crocodile-like head and a sail on its back.",
+    "description": "Meet the king of the river, a huge dinosaur about the size of a small house, with a ferocious crocodile-like head and a sail on its back. Usually feeds on fish.",
```
Comment (Qrox) [inline data/mods/DinoMod/monsters/dinosaur.json:148]:
```suggestion
    "description": "Meet the king of the river, a huge dinosaur about the size of a small house, with a ferocious crocodile-like head and a sail on its back.  Usually feeds on fish.",
```

**Comment by LyleSY** [comment]
Love the hint. The more complicated stuff will take me more time, I'll try and get to it tonight

**Review by LyleSY — CHANGES_REQUESTED** [review]
Spino spawn is in a weird place right now and needs a more complicated fix (see my long post on Discord). For now, please remove the wilderness spawn changes. Everything else makes sense.

**Review by LyleSY — APPROVED** [review]
Great stuff, congrats on your first!

**Pull request merged**